apps/website/src/pages/forms/web-dev-challenge/hackathon.astro

```diff
@@ -0,0 +1,255 @@
+---
+import { SignedIn, SignedOut, SignIn } from '@clerk/astro/components';
+import { getPersonById } from '@codetv/sanity';
+import { actions } from 'astro:actions';
+import Layout from '../../../layouts/default.astro';
+import '../../../styles/forms.css';
+
+const user = await Astro.locals.currentUser();
+
+const userDetails = (await getPersonById(
+	{ user_id: user?.id ?? '' },
+	{ useCdn: false },
+)) ?? { _id: undefined, bio: '', links: [] };
+
+const result = Astro.getActionResult(actions.forms.wdc.hackathon);
+
+if (result && !result.error) {
+	return Astro.redirect(`/forms/discord`);
+}
+---
+
+<Layout title="Web Dev Challenge Hackathon Submission">
+	<main>
+		<SignedIn>
+			<section class="form-wrapper">
+				<div class="instructions">
+					<h1>Web Dev Challenge Hackathon Submission</h1>
+					<p>
+						Thanks so much for taking part of the Web Dev Challenge Hackathon! This form will let you submit your hackathon project for [INSERT HACKATION HERE].
+					</p>
+				</div>
+
+				<div class="form">
+					<form method="POST" action={actions.forms.wdc.hackathon}>
+						<section class="form-section">
+                            <div class="section-header">
+								<h2>Contact Information</h2>
+							</div>
+							<div class="field">
+								<label for="email">Email</label>
+								<input
+									type="email"
+									name="email"
+									id="email"
+									class="input"
+									placeholder="you@example.com"
+									value={user?.primaryEmailAddress?.emailAddress ?? ''}
+									required
+								/>
+							</div>
+
+							<div class="field">
+								<label for="fullName"
+									>Full name</label
+								>
+								<input
+									type="text"
+									name="fullName"
+									id="fullName"
+									class="input"
+									placeholder="Full name"
+									required
+								/>
+							</div>
+						</section>
+
+						<section class="form-section">
+							<div class="section-header">
+								<h2>Project Submission</h2>
+							</div>
+
+							<div class="field">
+								<label for="githubRepo">Please provide a link to the GitHub repo where this web app's source code is published</label>
+								<p class="description">
+									Please ensure that all credentials are moved to environment variables before committing your code!
+								</p>
+								<input
+									type="url"
+									name="githubRepo"
+									id="githubRepo"
+									class="input"
+									placeholder="https://www.github.com/..."
+									pattern="https://www.github.com/.*"
```

Comment:
Something to think about: do we want to allow gitlab, etc.? We probably should. Need to think a bit about whether that presents any issues. Probably not?

Reply:
As long as it's a public link to the code's repository, I don't see the harm in allowing links to other repositories unless there are deeper integrations planned.